sw/host/cryptotest/testvectors/parsers/nist_cavp_rsa_parser.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Parser for converting NIST CAVP RSA test vectors to JSON.
+
+"""
+
+import argparse
+import sys
+import json
+import jsonschema
+
+from cryptotest_util import parse_rsp, str_to_byte_array
+
+SUPPORTED_SECURITY_LEVELS = [
+    2048,
+    3072,
+    4096,
+]
+
+HASH_NAME_MAPPING = {
+    "SHA256": "sha-256",
+    "SHA384": "sha-384",
+    "SHA512": "sha-512",
+}
+
+
+def parse_testcases(args) -> None:
+    raw_testcases = parse_rsp(args.src)
+    test_cases = list()
+
+    # NIST splits the rsp files into sections with named after (curve, hash
+    # algorithm) pairs
+    for section_name in raw_testcases.keys():
+        mod = int(section_name.split(" = ")[1])
+        if mod not in SUPPORTED_SECURITY_LEVELS:
+            continue
+        n = []
+        d = []
+        e = []
+        count = 0
+        for test_vec in raw_testcases[section_name]:
+            # The NIST vectors list the value of n on its own, and it
+            # applies to multiple test cases.
+            if "n" in test_vec:
```

Comment:
Once again, NIST doesn't have a common format for the RSP files. For RSA, they put in parameters on their own within groups that are supposed to apply to multiple test cases, and this was the cleanest way to handle that with the existing infrastructure.

Reply:
Edit: doing this was obsoleted by #21728, which this PR now depends on.